[WIP] Follow upstream

pocke opened the pull request:
ff064edf83e3df6d7f086c180f7a39103a925442...32d69d4b6ab30497d1ca7c0f16fd602329af7bdc
のコミットに日本語版も追従しました。

rrosenblum committed:
Make the example for trailing underscore variables clearer

lee-jon committed:
Prefer raise over fail for exceptions

Raise is more community supported, and the previous distinction between
fail and raise has not been adopted. This adds a new rule for the
preference and removed reference to fail.

Discussion: https://github.com/bbatsov/ruby-style-guide/issues/512

lee-jon committed:
Markdown fix - change four-backticks to three for consistency

bbatsov committed:
Merge pull request #515 from lee-jon/remove-extra-backticks

Markdown fix - change four-backticks to three for consistency

bbatsov committed:
Merge pull request #506 from rrosenblum/trailing_underscore_variables

Make the example for trailing underscore variables clearer

bbatsov committed:
Merge pull request #517 from lee-jon/remove-fail-raise-distinction

Prefer raise over fail

bbatsov committed:
Fix anchor name

ojab committed:
Add a rule about Hash#each_key & Hash#each_value

syndbg committed:
Add Ruby 2.3 squiggly heredocs usage tip

bbatsov committed:
Merge pull request #523 from syndbg/squiggly-heredocs

Add Ruby 2.3 squiggly heredocs usage tip

Commit:
Update README.md

For consistency. Hyphenated all but one 'multi-line'

bbatsov committed:
Merge pull request #524 from ojab/master

Add a rule about Hash#each_key & Hash#each_value

bbatsov committed:
Merge pull request #525 from jongaegler/patch-1

Missing '-'

etdev committed:
Fix minor typo in hash-each

bbatsov committed:
Merge pull request #529 from etdev/fix-typo-hash-key-value

Fix minor typo in hash-each

araslanov-e committed:
Update README.md

Fix result some_method

bbatsov committed:
Merge pull request #531 from araslanov-e/patch-1

Update README.md

cassioscabral committed:
Fix typo

bbatsov committed:
Merge pull request #533 from cassioscabral/patch-1

Fix typo

Neodelf committed:
Improve some examples and apply some formatting corrections

bbatsov committed:
Merge pull request #551 from Neodelf/additions_and_formatting

Some small additions and corrections of formatting

akowalz committed:
Fix indentation in naming section and typo in error subclass section

bbatsov committed:
Merge pull request #552 from akowalz/indentation_and_typo_fixes

Fix indentation in naming section and typo in error subclass section

bbatsov committed:
Replace the mention of Transmuter with Pandoc

Commit:
Update README.md (#562)

Change verb to plural.  ("underscores is" to "underscores are").

jessieay committed:
Avoid prefixing predicate methods with auxiliary verbs (#566)

* Redundant and inconsistent with Ruby core lib bool methods
* Description of auxiliary verbs: https://en.wikipedia.org/wiki/Auxiliary_verb
* Also works nicely with RSpec predicate matchers
* https://www.relishapp.com/rspec/rspec-expectations/v/2-4/docs/built-in-matchers/predicate-matchers

landongrindheim committed:
Hyphenate 'multi-line' consistently (#565)

Commit:
Gather rules about spaces around braces in one place (#560)

accessd committed:
Show results of using the splat operator (#534)

dpisarewski committed:
Add a type-checking rule for integer numbers (#564)

WizardOfOgz committed:
Change Array check splat(*) example from "good" to "bad": (#541)

Using the splat technique `[*object]` will _always_ instantiate a new Array object, whereas
`Array(object)` will return the same object it is given as an argument if that argument is already an Array.

Commit:
Remove duplicate section "pad-string-interpolation" (#568)

tejasbubane committed:
Add section for octal literals (#578)

Prefer `0o` over `0` and `0O` for octal literals.

tejasbubane committed:
Add rules for hexadecimal, binary and decimal literals (#579)

Related to #578

@bbatsov suggested to add rule for hex numbers. I went ahead and added
one for binary as well.

tejasbubane committed:
Add accessor and mutator method names conventions (#582)

Add a section to avoid using accessor and mutator method names starting
with `get_` and `set_`.

PanisSupraOmnia committed:
Improve minor grammar and clarity issues

PanisSupraOmnia committed:
Use "&mdash" instead of space-surrounded hyphen

Just looks a little better with proper punctuation

PanisSupraOmnia committed:
Fix formatting and attribution of quote

PanisSupraOmnia committed:
Change style of Yoda quote

DarkSideF committed:
Fix code example (#588)

Drenmi committed:
Add information about Time versus DateTime (#587)

As per the [documentation](http://ruby-doc.org/stdlib-2.3.0/libdoc/date/rdoc/DateTime.html),
`DateTime` should only be used in cases where historical calendar
reform needs to be taken into consideration. In all other cases,
using `Time` or `Date` is advisible.

Commit:
[Add] Vietnamese version (#577)

Commit:
Clarify "no-bang-bang" (#581)

Commit:
Gather rules of the use of parentheses in method invocations (#570)

DamirSvrtan committed:
Add randon numbers style guide

hediyi committed:
no-parens-if -> no-parens-around-condition

hediyi committed:
Rectify no-parens-around-condition

hedgesky committed:
Use new example in parentheses omitting section

gnapse committed:
Improve indentation

This particular code sample is not about indentation, but about the use of trailing commas in method arguments. So the standard indentation should be used, and it still conveys the same message without the added ugliness of the non-standard indentation previously used.

jdickey committed:
Proposing rule for referencing class methods in other class methods.

Using the example of a "service object" and its standard `.call` class
method that instantiates the class it's a member of, this proposer
suggests that class methods simply call other class methods, rather than
regurgitating the class name or even the `self` keyword. Many people,
including this proposer, tend to forget that `new` will (generally) Do
The Right Thing in such situations without decoration *because* the
method in `BasicObject` is overridden appropriately in `Class` and
elsewhere.

bbatsov committed:
Kill the rule suggesting the use of specific reduce block argument names

It's really debatable whether using names like `a` and `e` is a good
practice.

Commit:
Add rule when variable name has a number

bbatsov committed:
[Fix #574] Improve the name of a class used in an example

mentero committed:
Remove link to a e-mail regarding has_key? deprecation (#595)

abraaomiranda committed:
Mention the locale of the Portuguese translation (#604)

adrianomitre committed:
[Fix #511] Clarify when the usage of self is necessary (#530)

davearonson committed:
Clarify why 'and' and 'or' are evil (#489)

martenlienen committed:
Remove an unnecessary non-nil check (#482)

They are discouraged in another part of the style guide [1].

[1] https://github.com/bbatsov/ruby-style-guide#no-non-nil-checks

johana-star committed:
Clarify when to use percent literals

Connects #284

backus committed:
Add guidelines for how to use magic comments

dpisarewski committed:
Prefer square brackets over round brackets for array literals (#563)

riseshia committed:
Remove some trailing whitespace (#620)

Drenmi committed:
Add a section on mixin grouping

This change adds a suggested rule on mixin grouping. Some rationales for
this could be:

 - Easier to see the number of dependencies for a `class` or `module`.
 - Easier and less error prone to change or remove a mixin.

Mostly, however, it is for consistency.

pocke committed:
Merge commit '652d73f0e68f9a87984db3fdb854f56d499a6843' into follow-upstream

pocke committed:
Follow 652d73f0e68f9a87984db3fdb854f56d499a6843

pocke committed:
Merge commit '91a60a36ef18702366f7c63a0b241ecf8d8935fa' into follow-upstream

hayfever committed:
Fix a typo (#626)

pocke committed:
Follow 91a60a36ef18702366f7c63a0b241ecf8d8935fa

pocke committed:
Merge commit '12b597f40b42ff88821269c3aed2ceda6e2de604' into follow-upstream

pocke committed:
Follow 12b597f40b42ff88821269c3aed2ceda6e2de604

pocke committed:
Merge commit '91217d4e1b45c33d2e20759194ace3f359cf11e1' into follow-upstream

pocke committed:
Follow 91217d4e1b45c33d2e20759194ace3f359cf11e1

pocke committed:
Merge commit '2dcf5f9f5d13b212782cfa7bdac1b6d05ee56d0e' into follow-upstream

pocke committed:
Merge commit '815a0b9208cfaf4d206750a4cd0ebd28c7f1d636' into follow-upstream

pocke committed:
Follow 815a0b9208cfaf4d206750a4cd0ebd28c7f1d636

pocke committed:
Merge commit '122f8178978cf0c45dd2d6cef80f07e36526afa7' into follow-upstream

pocke committed:
Follow 122f8178978cf0c45dd2d6cef80f07e36526afa7

pocke committed:
Merge commit '18be9b3f55bf8b32db1188ebad3601d25f1c4f4c' into follow-upstream

pocke committed:
Merge commit 'e10cd1054ca7abeadca0a33a190f76e1552c7cae' into follow-upstream

pocke committed:
Follow 88bfea59859c3d7a1fd0d47624e1ba96fc00d168

pocke committed:
Merge commit '1c8e89d02ab3766186ebbe7d01c87e78642318ca' into follow-upstream

pocke committed:
Merge commit 'ef0852f0c2f949950a520aec8bd83a90de4b79e9' into follow-upstream

pocke committed:
Follow ef0852f0c2f949950a520aec8bd83a90de4b79e9

pocke committed:
Merge commit '06d503f6bef11a18d0a1e94e84addae69268fdf0' into follow-upstream

pocke committed:
Merge commit 'd3fdf34d26767709c3725d3fffc87dbc4c829aad' into follow-upstream

pocke committed:
Follow d3fdf34d26767709c3725d3fffc87dbc4c829aad

pocke committed:
Merge commit 'c8e5ac8b5489a26cec403c4b570568f8d40e25fd' into follow-upstream

pocke committed:
Follow f88e1de9367935c9561b5e0b86fa093827dc1fe9

pocke committed:
Follow c0e3a698d98fc3eeaa58776e1d91a64df5a4089d

pocke committed:
Follow bd6f8a2af1b3513e7e3053136fcb1a9fddea78ed

pocke committed:
Follow 39d176b05a692b83a247a5f41b8ff3003010f82b

pocke committed:
Follow 49ee417727f489ce14e62fac4b0b3ccd24a0f40e

pocke committed:
Follow d3baf587eacf2401301385d982e6dfdfbfc748a9

pocke committed:
Follow 496afebd650d9277917dbfa7ebb85d12a7c5e084

pocke committed:
Follow 3dedf5fe6cd8023a23b8cfe647d4d1a593117ddd

pocke committed:
Follow 4020c0bcd30190d521073fa2e2260b94eaee94c8

pocke committed:
Follow 3344546f98e532475ab7b02f1749cc09554e7361

pocke committed:
Follow 7fd8d5d4311047f52db34d676eed0ab421c33e34

pocke committed:
Follow 6c204fe90c4c0c02798da36a94078f9cb0f379ef

pocke committed:
Follow 350b0271e4ced8f97a585986784d9044d2fae183

pocke committed:
Follow 4c43ea243779da5e15ffd2b07126627ef4ce6327

pocke committed:
Follow a082855580b023fbbe018a07cf2153e3726e5f20

pocke committed:
Follow 3e3d3165105565ef97fdddd4300c9774f1b01bf7

pocke committed:
Follow 8a837aa43fda51786d8e14d9013e7f6ae15ab24b

pocke committed:
Follow 6da08f1bfe3948ace3b6a5d073155192a4b6b07a

pocke committed:
Follow 44758db39574d6fd964fd773612302b02a824a30

pocke committed:
Follow 8a483345e55789125947accef2e6932e536c5739

pocke committed:
Follow 09eb304929dda8e31a31ae21cc3ddbc2d2b24fc8

pocke committed:
Follow b2a8afd834737e92b4120316b407e7e6f99bf0d4

pocke committed:
Follow 596d7b60fa610d871bd63d27f27f581afa780558

pocke committed:
Follow df41e1ef3d6223883408f5d50da87e9aae60a765

pocke committed:
Follow e12cf586a51c6a10b657476905a126db40e7b123

pocke committed:
Follow 03f26278666c145590638727587a24261f015da6

pocke committed:
Follow 65a4529844536a91b0a31be073edffcefb8c91ca

pocke committed:
Follow 1c2928bbe4cf6513f41ac27e08c05d5a138cee56

pocke committed:
Follow 141a70623fb89c4af0e32986da8871fa3484d116

pocke committed:
Follow dedd33a68e86335b1dd7c1e3766d5bf7d5aa47d3

pocke committed:
Follow f981287d709e03b3cd2f7ff856d1a67412735954

pocke committed:
Follow 37960a0200e49a609eae8e29b6f728c32b0963ac

pocke committed:
Follow ac903faa81663d1037a3968f805ee8219e7f8c38

pocke committed:
Follow 8d88659450aeda27007d7959c1259be43bc5f231

pocke committed:
Follow c07bf9198642bc13c93bb576d0df6e3907208789

pocke committed:
Follow 32d69d4b6ab30497d1ca7c0f16fd602329af7bdc

shyouhei commented:
ありがとうございます。しかしちょっと大きいので、読む時間をください。

shyouhei commented on README.ja.md:
```diff
@@ -703,32 +803,43 @@ Rubyコミュニティ内でもスタイルについての統一見解が存在
-  アンダースコア変数は左辺にsplat変数を定義するときには必要です。
-  その場合、splat変数はアンダースコアではありえないです。
+  名前付きアンダースコア変数はコンテキストを提供するため、 アンダースコア変数
+  より優先される必要があります。
```
ここはちょっと疑問かも。「優先される」というのが、すぐには意味が取れませんでした。下の例まで読めばようやく、 `*_` だけじゃなくて `*_name` みたいにしておけという話なのがわかります。

ちょっと長くなるけど、自分なら「アンダースコア変数は名前付きにしておくほうがコンテキストを明示できるので、名前をつけておいたほうが良いです」かなあ。

shyouhei commented on README.ja.md:
```diff
@@ -881,6 +992,10 @@ Rubyコミュニティ内でもスタイルについての統一見解が存在
   `!!`は避けましょう。
 <sup>[[link](#no-bang-bang)]</sup>
 
+  `!!`は値をブーリアンに変換しますが、制御式の条件で明示的な変換は必要ありません。
```
どうやら、「ブーリアン」という書き方はここにしか出てこないようです。他はbooleanになっていました。

スタイルガイドの文章なので、それ自体のスタイルは統一されているべきかと思います。

pocke replied:
承知しました、boolean に統一します。

shyouhei commented on README.ja.md:
````diff
@@ -642,6 +674,74 @@ Rubyコミュニティ内でもスタイルについての統一見解が存在
+  temperance = Person.new('Temperance', 30)
+  ```
+
+  以下の場合は括弧を省略します
````
元の文("Only omit parentheses for")の解釈が分かれるところだと思うのですが、この「以下の場合」では括弧は必ず省略すべきだと思いますか？それとも省略しても良い程度の話だと思いますか？

どちらともとれるのですが、いまの訳だと必ず省略するように読めますね。ただ、最後のputsの例で括弧を入れることは自分としてはよくあるので、自分の解釈は「省略しても良い」なのではという気がしています。

pocke replied:
この訳について考え直してみましたが、私の感覚では「必ず省略」「省略しても良い」のどちらも適切でないように思えます。

仰るとおり`puts`の例で括弧を入れることは考えられるので、「必ず省略」という解釈は不自然に思えます。

対して`'test'.upcase()` のような引数のないメソッドは私の感覚では必ず括弧を省略すべき、という解釈のほうが適合していて、「省略してもよい」の解釈は合わないように思えます。
オリジナルのruby-style-guideの作者と同作者の RuboCop でも、このケースは「必ず省略すべき」となっているので、引数の内メソッドの場合はこちらの解釈がより適していると思います。
http://rubocop.readthedocs.io/en/latest/cops_style/#stylemethodcallwithoutargsparentheses

ですので、この件に関してはオリジナルの ruby-style-guide に Issue をあげようと考えています。

- 「引数のないメソッド」「内部DSLに含まれるメソッド(例えばRake, Rails, RSpec)」は「括弧を必ず省略すべき」
- 「Rubyで「キーワード」と認識されているメソッド」の内、`attr_reader` のような宣言的なものは「括弧を必ず省略すべき」
- 「Rubyで「キーワード」と認識されているメソッド」の内、`puts` のようなトップレベルに生えているメソッドは「省略しても良い」

というふうに、必ず省略すべきかを明記するようにオリジナルの方を修正するよう提案しようと思います

pocke replied:
https://github.com/bbatsov/ruby-style-guide/issues/635 Issue を上げました

shyouhei commented on README.ja.md:
```diff
@@ -1691,25 +1763,25 @@ Rubyコミュニティ内でもスタイルについての統一見解が存在
   paths.each { |path| do_something(path) }
 
-  # 良い例
+  # 悪い例 (常に新しい`Array`のインスタンスを生成します)
```
この `` `Array` `` は変じゃないですか？ ```` ``` ```` の中なんだし。

pocke replied:
Array とただ書いているとわかりづらいと思ってこうしましたが、見返すと`` ` `` はない方がよさそうですね

shyouhei commented on README.ja.md:
```diff
@@ -2023,6 +2123,44 @@ Rubyコミュニティ内でもスタイルについての統一見解が存在
 <sup>[[link](#bool-methods-qmark)]</sup>
 
+* <a name="bool-methods-prefix"></a>
+  述語メソッドに `is`、` does`、 `can`のような補助動詞をつけるべきではありません。
```
"auxiliary verbs" のことを日本語では「助動詞」といいます。

pocke commented:
ありがとうございます。 かなり大きくて申し訳ないですがよろしくお願いします 🙏

shyouhei commented on README.ja.md:
```diff
@@ -2251,6 +2451,24 @@ Rubyコミュニティ内でもスタイルについての統一見解が存在
+
+  # 良い例
+  class Person
+    # 複数のミックスインは分割された文になります
```
「なります」より「します」が好きです。日本語ではあまり無生物主語を使いません。

shyouhei commented on README.ja.md:
```diff
@@ -2672,6 +2922,34 @@ Rubyコミュニティ内でもスタイルについての統一見解が存在
+* <a name="class-and-self"></a>
+  クラス(またはモジュール)メソッドが他のメソッドを呼び出す際は、
+  `self`または自分自身のクラス名の後に`.`を続けて使用しないようにしましょう。
+  これは、「サービスクラス」やそれが関数であるかのように扱われる他の同様の概念でよく見られます。
```
この文は正しく訳せていると思うのですが、ちょっと入り組んでいて、読みづらかったです。

たとえば「これは、「サービスクラス」やそれと類似の、クラスがあたかも関数であるかのように扱われるコンセプトの場合でよく見られます」だとどうでしょう。

shyouhei commented on README.ja.md:
```diff
@@ -3172,33 +3466,52 @@ Rubyコミュニティ内でもスタイルについての統一見解が存在
 
+  * <a name="random-numbers"></a>
+  オフセットを持つ乱数を生成する場合は、整数の代わりに範囲リテラルを使用しましょう。
+  意図が明確であるためです。サイコロの役割をシミュレートすることを想像してください。
```
「オフセットを持つ乱数」は違うと思います。

```
( Prefer to use 
    ( ranges ※)
  instead of
    ( integers with offsets ) , ...)

※:     when generating random numbers
```

こういう構造の文だと思います。「整数とオフセット」のかわりに「範囲」を使いましょうということ。

shyouhei replied:
「サイコロの役割をシミュレートすることを想像してください」は下の例の説明ですから、「。」ではなく原文の通り「：」になっていたほうがよさそうです

shyouhei commented on README.ja.md:
````diff
@@ -3353,6 +3666,65 @@ Rubyコミュニティ内でもスタイルについての統一見解が存在
   ```
 
+* <a name="squiggly-heredocs"></a>
+  インデントされた複数行の文字列には、Ruby 2.3 の、インデントされた
````
原文にある"nicely"のニュアンスが抜けてそうです。

shyouhei commented on README.ja.md:
```diff
@@ -3353,6 +3666,65 @@ Rubyコミュニティ内でもスタイルについての統一見解が存在
 
+* <a name="squiggly-heredocs"></a>
+  インデントされた複数行の文字列には、Ruby 2.3 の、インデントされた
+  ヒアドキュメントを使用する必要があります。
```
「必要があります」は口調が強いかなと思いました。「使いましょう」でよいのでは。

shyouhei commented on README.ja.md:
```diff
@@ -3353,6 +3666,65 @@ Rubyコミュニティ内でもスタイルについての統一見解が存在
+<sup>[[link](#time-now)]</sup>
+
+* <a name="no-datetime"></a>
+  改暦の説明の必要がある場合を除いて、`DateTime`を使用しないようにしましょう。
```
この"need to account for"は「説明」じゃなくて「考慮」だと思います。「改暦を考慮する必要がある場合を除いて〜」

shyouhei reviewed (changes requested):
読みました。原文の変更はもれなく反映されており、一部に修正の必要がありますが、大部分はそのまま採用できそうです。ありがとうございます。

pocke committed:
Review 点を修正

- s/ブーリアン/boolean/
- Trim backtick
- s/補助動詞/助動詞/
- s/なります/します
- s/使用する必要があります/使いましょう
- s/説明の/考慮する

pocke committed:
レビュー点を修正

For the following styles

- trailing-underscore-variables
- class-and-self
- random-numbers
- squiggly heredocs

pocke commented:
@shyouhei レビューありがとうございます。遅くなってしまい申し訳ないです。
https://github.com/fortissimo1997/ruby-style-guide/pull/25#discussion_r99977387 の「括弧を必ず省略するか」の点以外は修正しました。
この点はコメントにもあるとおり、オリジナルのruby-style-guideの方にIssueをあげようと思います

shyouhei reviewed (approved):
👍

shyouhei commented:
修正は妥当だと思います。括弧の件はupstreamの結論を待ってからでいいと思いますので、現状の修正は取り込みますね。ありがとうございました。
